components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java

```diff
@@ -1014,13 +1029,61 @@ private List<String> patchRemoveOpDataGrabber(String query, String roleId, Strin
      */
     private List<User> getUsersFromRoleId(String roleId) throws OrganizationManagementServerException {
 
+        boolean isUserResidentOrgIDColumnExists = checkUserResidentOrgIDColumnExists();
+        String requestInvokingOrgId = getOrganizationId();
+
         NamedJdbcTemplate namedJdbcTemplate = getNewTemplate();
         try {
-            return namedJdbcTemplate.withTransaction(template -> template.executeQuery(GET_USERS_FROM_ROLE_ID,
-                    (resultSet, rowNumber) -> new User(resultSet.getString(DB_SCHEMA_COLUMN_NAME_UM_USER_ID)),
-                    namedPreparedStatement ->
-                            namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_UM_ROLE_ID, roleId)));
-        } catch (TransactionException e) {
+            List<User> users;
+
+            if (isUserResidentOrgIDColumnExists) {
+                users = namedJdbcTemplate.withTransaction(template ->
+                        template.executeQuery(GET_USERS_FROM_ROLE_ID_WITH_ORG, (resultSet, rowNumber) -> {
+                            User user = new User(resultSet.getString(DB_SCHEMA_COLUMN_NAME_UM_USER_ID));
+                            user.setUserResidentOrgId(resultSet.getString(DB_SCHEMA_COLUMN_NAME_USER_RES_ORG_ID));
+                            user.setUserResidentOrgName(resultSet.getString(DB_SCHEMA_COLUMN_NAME_UM_ORG_NAME));
+                            return user;
+                            }, namedPreparedStatement ->
+                                namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_UM_ROLE_ID, roleId)));
+
+                // Resolving and saving the user's resident organization ID if it is not available in the DB.
+                for (User user : users) {
+                    if (StringUtils.isBlank(user.getUserResidentOrgId())) {
+                        Optional<String> resolvedUserResidentOrgId = getOrganizationUserResidentResolverService()
+                                .resolveResidentOrganization(user.getId(), requestInvokingOrgId);
+                        if (resolvedUserResidentOrgId.isPresent()) {
+                            user.setUserResidentOrgId(resolvedUserResidentOrgId.get());
+                            user.setUserResidentOrgName(getOrganizationManager()
+                                    .getOrganizationNameById(user.getUserResidentOrgId()));
```

Comment:
Is there any special requirement to access this from the settled value to the user?
```suggestion
                                    .getOrganizationNameById(resolvedUserResidentOrgId.get()));
```

Reply:
Resolved with https://github.com/wso2-extensions/identity-organization-management/pull/200/commits/92ac8796ca6cfce6426beeef5426b69e141fbee7